src/TransportClient/Curl/HttpClient.php

```diff
@@ -86,4 +90,50 @@ public function post(string $url, ?array $payload): TransportClientResponse
             return new TransportClientResponse($httpStatus, $response);
         }
     }
+
+    private function readVersionFile()
```

Comment:
@MisterBugurt, well done  👍 
Just a few things that could be adjusted:

1. Consider moving this logic to Utils/EnvHelper.php and making it a static method. That way it can be used from any place like `EnvHelper::getVersion()` 
2. If the .version file is missing, it should not raise an exception but rather fail silently: the request should be sent as usual, just without the `X-API-Version` header